spec/models/application_spec.rb

```diff
@@ -1 +1,33 @@
-require "rails_helper"
+require 'rails_helper'
+
+RSpec.describe Application, type: :model do
+  describe 'relationships' do
+    it { should have_many(:pet_applications) }
+    it { should have_many(:pets, through: :pet_applications) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_presence_of(:adress) }
+    it { should validate_presence_of(:city) }
+    it { should validate_numericality_of(:state) }
+    it { should validate_numericality_of(:zip) }
+    it { should validate_numericality_of(:description) }
+  end
+
+  before(:each) do
+    @shelter_1 = Shelter.create(name: 'Aurora shelter', city: 'Aurora, CO', foster_program: false, rank: 9)
+    @shelter_2 = Shelter.create(name: 'RGV animal shelter', city: 'Harlingen, TX', foster_program: false, rank: 5)
+
+    @pet_1 = @shelter_1.pets.create(name: 'Mr. Pirate', breed: 'tuxedo shorthair', age: 5, adoptable: false)
+    @pet_2 = @shelter_1.pets.create(name: 'Clawdia', breed: 'shorthair', age: 3, adoptable: true)
+
+    @application1 = Application.create!(name: 'John Doe', address: '123 Elm St', city: 'Denver', state: 'CO',
+                                        zip: '12345', description: 'Looking for a friendly dog', status: 'In Progress')
+  end
+
+  describe 'class methods' do
```

Comment:
I'll have to flush out these model tests once we write more methods in the models.